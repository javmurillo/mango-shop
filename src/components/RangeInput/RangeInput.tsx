import React, { Component } from 'react';
import styled from 'styled-components';
import { RangeInputProps } from './range-input-props.interface';

const StyledInput = styled.input`
  border: none;
  text-align: right;
  outline: none;
  width: 50px;
  font-weight: bold;
`;

const StyledInputWrapper = styled.div`
  display: flex;
`;

const StyledEuro = styled.span`
  font-weight: bold;
`;
export default class RangeInput extends Component<RangeInputProps> {
  render(): JSX.Element {
    return (
      <StyledInputWrapper style={{ marginRight: '2rem' }}>
        <StyledInput
          type="number"
          value={this.props.value}
          onChange={ev => {
            this.props.onChange(ev, this.props.rangeKey);
          }}
          disabled={this.props.disabled}
        />
        <StyledEuro>â‚¬</StyledEuro>
      </StyledInputWrapper>
    );
  }
}
