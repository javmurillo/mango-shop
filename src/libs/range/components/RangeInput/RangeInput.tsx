import React from 'react';
import styled from 'styled-components';
import { RangeInputProps } from './range-input-props.interface';

const StyledInput = styled.input`
  border: none;
  text-align: right;
  outline: none;
  width: 50px;
  font-weight: bold;
`;

const StyledInputWrapper = styled.div`
  display: flex;
`;

const StyledEuro = styled.span`
  font-weight: bold;
`;

/**
 * RangeIpunt component. Cursor will show as not allowed if the input is disabled.
 * Each change will update its parent state.
 * @param props RangeInputProps
 */
export const RangeInput = (props: RangeInputProps) => {
  return (
    <StyledInputWrapper>
      <StyledInput
        type="number"
        value={props.value}
        onChange={ev => {
          props.onChange(ev, props.rangeKey);
        }}
        style={props.disabled ? { cursor: 'not-allowed' } : {}}
        disabled={props.disabled}
      />
      <StyledEuro style={{ marginRight: '1rem' }}>â‚¬</StyledEuro>
    </StyledInputWrapper>
  );
};
