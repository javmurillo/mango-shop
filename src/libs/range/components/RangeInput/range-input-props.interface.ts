export interface RangeInputProps {
  value?: number;
  onChange?: (
    event: React.ChangeEvent<HTMLInputElement>,
    key: 'start' | 'end'
  ) => void;
  rangeKey: 'start' | 'end';
  disabled?: boolean;
}
