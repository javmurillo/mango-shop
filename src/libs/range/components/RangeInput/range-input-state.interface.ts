export interface RangeInputState {
  value: string;
}
