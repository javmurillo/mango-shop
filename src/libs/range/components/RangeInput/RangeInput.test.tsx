import { cleanup, render } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import RangeInput from './RangeInput';
import TestRenderer from 'react-test-renderer';

afterEach(cleanup);

const setup = () => {
  const utils = render(
    <RangeInput
      value={50}
      max={100}
      min={0}
      rangeKey="start"
      ariaLabel="range-input"
    />
  );
  const input = utils.getByLabelText('range-input');
  return {
    input,
    ...utils,
  };
};

describe('<RangeInput/> tests', () => {
  test('renders correctly', () => {
    const { input } = setup();

    expect(input).toBeInTheDocument();
  });

  test('value cant exceed max', () => {
    const { input } = setup();

    userEvent.type(input, '10');

    expect(input).toHaveValue(50);
  });

  test('should allow numbers', () => {
    const { input } = setup();

    userEvent.clear(input);
    userEvent.type(input, '10');

    expect(input).toHaveValue(10);
  });

  test('should not allow letters nor symbols', () => {
    const { input } = setup();

    userEvent.clear(input);
    userEvent.type(input, 'testâ‚¬');

    expect(input).toHaveValue(null);
  });

  test('should allow to delete input', () => {
    const { input } = setup();

    userEvent.type(input, '{backspace}');

    expect(input).toHaveValue(5);
  });

  test('should be disable', () => {
    const input = render(
      <RangeInput
        value={50}
        max={100}
        min={0}
        rangeKey="start"
        ariaLabel="range-input"
        disabled={true}
      />
    ).getByLabelText('range-input');

    expect(input).toBeDisabled();
  });

  test('shoud match snapshot', () => {
    const testRendererJson = TestRenderer.create(
      <RangeInput
        value={50}
        max={100}
        min={0}
        rangeKey="start"
        ariaLabel="range-input"
      />
    ).toJSON();
    expect(testRendererJson).toMatchSnapshot();
  });
});
