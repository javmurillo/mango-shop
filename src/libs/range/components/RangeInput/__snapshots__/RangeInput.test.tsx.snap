// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RangeInput/> tests shoud match snapshot 1`] = `
<div
  className="sc-gsTCUz ywWHk"
>
  <input
    aria-label="range-input"
    className="sc-bdfBwQ iwQBvM"
    min="0"
    onChange={[Function]}
    style={Object {}}
    type="number"
    value="50"
  />
  <span
    className="sc-dlfnbm jjMaay"
  >
    â‚¬
  </span>
</div>
`;
